controllers/gce/controller/cluster_manager.go

```diff
@@ -170,6 +182,25 @@ func (c *ClusterManager) Checkpoint(lbs []*loadbalancers.L7RuntimeInfo, nodeName
 	return nil
 }
 
+func (c *ClusterManager) CreateInstanceGroups(servicePorts []backends.ServicePort) ([]*compute.InstanceGroup, error) {
+	var igs []*compute.InstanceGroup
+	var err error
+	for _, p := range servicePorts {
+		igs, _, err = instances.CreateInstanceGroups(c.instancePool, c.ClusterNamer, p.Port)
```

Comment:
Shouldn't we be appending to igs?

Reply:
CreateInstanceGroups creates instance groups in all zones and then adds the given named port to it.
Each time CreateInstanceGroups is called for a port, it always returns the same set of all igs. So we dont need append, just return the output of any call.

Ideally, we should call CreateInstanceGroups only the first time and then call AddNamedPort subsequent times, but that interface is not exposed yet. Will add a TODO and a comment explaining this.

Reply:
Added the comment and TODO